src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -4321,7 +4527,22 @@ the (Stream step state) = go state
 
 {-# INLINE runFold #-}
 runFold :: (Monad m) => Fold m a b -> Stream m a -> m b
-runFold (Fold step begin done) = foldlMx' step begin done
+runFold (Fold fstep begin done) (Stream step state) =
+    begin >>= \x -> go SPEC x state
+  where
+    -- XXX !acc?
+    {-# INLINE_LATE go #-}
+    go !_ acc st = acc `seq` do
+        r <- step defState st
+        case r of
+            Yield x s -> do
+                acc' <- fstep acc x
+                case acc' of
+                    FL.Done b -> return b
+                    FL.Done1 b -> return b
+                    FL.Partial acc'' -> go SPEC acc'' s
```

Comment:
we can call this `acc1`. I prefer names with number suffixes instead of prime. The reasons are written in the style guide.